### ISSUE TITLE
encode_packed produces wrong output

### ISSUE THREAD

**Issue by krisbitney** [gakonst/ethers-rs#2257]
**Version**
2.0.0

**Platform**
Darwin (mac os) M1 apple cpu

**Description**
the abi.encode_packed function does not produce the output described in the Solidity spec.

A negative i16 `-1` seems to be interpreted by ethers-rs as a uint256.
```ts
types: [ "int16", "uint48" ]
values: [ "-1", "12" ]
ethers-js: "0xffff00000000000c"
ethers-rs: "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0c"
```

Another example where the packages disagree:
```ts
types: [ "uint256", "uint256" ]
values: [ "8", "16" ]
ethers-js: "0x00000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000010"
ethers-rs: "0x0810"
```

**Comment by DaniPopes** [comment]
Please see #2225 and the docs of the function.

**Comment by krisbitney** [comment]
> Please see #2225 and the docs of the function.

Thanks. I just reviewed these. It sounds like people are not concerned that the ethers-rs implementation differs from the Solidity implementation, because the limitations of the ethers-rs implementation are known (e.g. can't distinguish between uint sizes). Is that an accurate summary?

**Comment by DaniPopes** [comment]
Yeah, the biggest limitation is that we are delegating most of the ABI encoding functionality to an external crate (`ethabi`), which is fine until we want to add new functionality to it, like `encode_packed`, which requires a different API.. I think this is a goal for 3.0 tho
